### ISSUE TITLE
[test] issue during the Debian build process

### ISSUE THREAD

**Issue by picca** [silx-kit/silx#1033]
Hello, I unactviated the OPENCL tests and I got this error.

```
WARNING:run_tests:Test silx 0.5.0 from /<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/silx
WARNING:silx.test.test_sx:silx.sx tests disabled (DISPLAY env. variable not set)
WARNING:fabioimage:PIL is not installed ... trying to do without
WARNING:tifimage:PIL is not installed ... trying to do without
WARNING:bruker100image:PIL is not installed ... trying to do without
WARNING:xsdimage:lxml library is probably not part of your python installation: disabling xsdimage format
WARNING:silx.opencl:Use of OpenCL has been disables from environment variable: SILX_OPENCL=0
Traceback (most recent call last):
  File "run_tests.py", line 369, in <module>
    unittest.defaultTestLoader.loadTestsFromNames(options.test_name))
  File "/usr/lib/python2.7/unittest/loader.py", line 130, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/usr/lib/python2.7/unittest/loader.py", line 115, in loadTestsFromName
    test = obj()
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/silx/test/__init__.py", line 52, in suite
    from ..image import test as test_image
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/silx/image/test/__init__.py", line 33, in <module>
    from . import test_medianfilter
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/silx/image/test/test_medianfilter.py", line 34, in <module>
    from silx.image import medianfilter
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/silx/image/medianfilter.py", line 36, in <module>
    from silx.opencl import medfilt as medfilt_opencl
  File "/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build/silx/opencl/__init__.py", line 46, in <module>
    from .common import *
AttributeError: 'module' object has no attribute 'mf'
E: pybuild pybuild:283: test: plugin custom failed with: exit code=1: env PYTHONPATH=/<<PKGBUILDDIR>>/.pybuild/pythonX.Y_2.7/build WITH_QT_TEST=False SILX_OPENCL=False SILX_TEST_LAW_MEM=True python2.7 run_tests.py -v
dh_auto_test: pybuild --test -i python{version} -p 2.7 -s custom "--test-args=env PYTHONPATH={build_dir} WITH_QT_TEST=False SILX_OPENCL=False SILX_TEST_LAW_MEM=True {interpreter} run_tests.py -v" returned exit code 13
```

Cheers

**Comment by picca** [comment]
Ok looking at the code of common, mf is not defined when the SILX_OPENCL=0 or False

```
if os.environ.get("SILX_OPENCL") in ["0", "False"]:
    logger.warning("Use of OpenCL has been disables from environment variable: SILX_OPENCL=0")
    pyopencl = None
    <mf should be define here also>
else:
    try:
        import pyopencl
    except ImportError:
        logger.warning("Unable to import pyOpenCl. Please install it from: http://pypi.python.org/pypi/pyopencl")
        pyopencl = None
        class mf(object):
            WRITE_ONLY = 1
            READ_ONLY = 1
            READ_WRITE = 1
    else:
        import pyopencl.array as array
        mf = pyopencl.mem_flags
```

**Comment by picca** [comment]
Another problem already saw on pyFAI :)

```
WARNING:run_tests:Test suite failed
usage: prog [--version|--help] <command> [<args>]

The prog commands are:
   foo        A command
   help       Show help of the following command

See 'prog help <command>' to read about a specific subcommand
('Unknown command: %s', 'foo2')
usage: prog [--version|--help] <command> [<args>]

The prog commands are:
   foo        A command
   help       Show help of the following command

See 'prog help <command>' to read about a specific subcommand
usage: run_tests.py [--version|--help] <command> [<args>]

The run_tests.py commands are:
   help       Show help of the following command

See 'run_tests.py help <command>' to read about a specific subcommand
usage: run_tests.py [--version|--help] <command> [<args>]

The run_tests.py commands are:
   help       Show help of the following command

See 'run_tests.py help <command>' to read about a specific subcommand
run_tests.py version None
usage: run_tests.py help [-h] [command]

Display help information about run_tests.py

positional arguments:
  command     Command in which aving help

optional arguments:
  -h, --help  show this help message and exit
usage: run_tests.py help [-h] [command]

Display help information about run_tests.py

positional arguments:
  command     Command in which aving help

optional arguments:
  -h, --help  show this help message and exit
Unknown command: foo2
usage: prog [--version|--help] <command> [<args>]

The prog commands are:
   foo        A command
   help       Show help of the following command

See 'prog help <command>' to read about a specific subcommand
usage: prog [--version|--help] <command> [<args>]

The prog commands are:
   foo        A command
   help       Show help of the following command

See 'prog help <command>' to read about a specific subcommand
('Unknown command: %s', 'foo2')
usage: prog [--version|--help] <command> [<args>]

The prog commands are:
   foo        A command
   help       Show help of the following command

See 'prog help <command>' to read about a specific subcommand
usage: run_tests.py [--version|--help] <command> [<args>]

The run_tests.py commands are:
   help       Show help of the following command

See 'run_tests.py help <command>' to read about a specific subcommand
usage: run_tests.py [--version|--help] <command> [<args>]

The run_tests.py commands are:
   help       Show help of the following command

See 'run_tests.py help <command>' to read about a specific subcommand
run_tests.py version None
usage: run_tests.py help [-h] [command]

Display help information about run_tests.py

positional arguments:
  command     Command in which aving help

optional arguments:
  -h, --help  show this help message and exit
usage: run_tests.py help [-h] [command]

Display help information about run_tests.py

positional arguments:
  command     Command in which aving help

optional arguments:
  -h, --help  show this help message and exit
Unknown command: foo2
usage: prog [--version|--help] <command> [<args>]

The prog commands are:
   foo        A command
   help       Show help of the following command

See 'prog help <command>' to read about a specific subcommand
```

**Comment by vallsv** [comment]
The log with application output should be fixed now if you use the last master. But the behaviour do not look at the same according to different operating system.

**Comment by payno** [comment]
this `AttributeError: 'module' object has no attribute 'mf'` issue have been fixed by commit 47996542

**Comment by t20100** [comment]
closing as both issues should be fixed
